python_modules/libraries/dagster-sigma/dagster_sigma/translator.py

```diff
@@ -71,46 +76,54 @@ def __init__(self, context: SigmaOrganizationData):
     def organization_data(self) -> SigmaOrganizationData:
         return self._context
 
-    def get_workbook_key(self, data: SigmaWorkbook) -> AssetKey:
-        return AssetKey(_clean_asset_name(data.properties["name"]))
-
-    def get_workbook_spec(self, data: SigmaWorkbook) -> AssetSpec:
-        metadata = {
-            "dagster_sigma/web_url": MetadataValue.url(data.properties["url"]),
-            "dagster_sigma/version": data.properties["latestVersion"],
-            "dagster_sigma/created_at": MetadataValue.timestamp(
-                isoparse(data.properties["createdAt"])
-            ),
-        }
-        datasets = [self._context.get_datasets_by_inode()[inode] for inode in data.datasets]
-        return AssetSpec(
-            key=self.get_workbook_key(data),
-            metadata=metadata,
-            kinds={"sigma"},
-            deps={self.get_dataset_key(dataset) for dataset in datasets},
-            owners=[data.owner_email] if data.owner_email else None,
-        )
-
-    def get_dataset_key(self, data: SigmaDataset) -> AssetKey:
-        return AssetKey(_clean_asset_name(data.properties["name"]))
-
-    def get_dataset_spec(self, data: SigmaDataset) -> AssetSpec:
-        metadata = {
-            "dagster_sigma/web_url": MetadataValue.url(data.properties["url"]),
-            "dagster_sigma/created_at": MetadataValue.timestamp(
-                isoparse(data.properties["createdAt"])
-            ),
-            **TableMetadataSet(
-                column_schema=TableSchema(
-                    columns=[TableColumn(name=column_name) for column_name in sorted(data.columns)]
-                )
-            ),
-        }
-
-        return AssetSpec(
-            key=self.get_dataset_key(data),
-            metadata=metadata,
-            kinds={"sigma"},
-            deps={AssetKey(input_name.lower().split(".")) for input_name in data.inputs},
-            description=data.properties.get("description"),
-        )
+    def get_asset_key(self, data: Union[SigmaDataset, SigmaWorkbook]) -> AssetKey:
+        """Get the AssetKey for a Sigma object, such as a workbook or dataset."""
+        return AssetKey(_coerce_input_to_valid_name(data.properties["name"]))
+
+    def get_asset_spec(
+        self, asset_key: AssetKey, data: Union[SigmaDataset, SigmaWorkbook]
+    ) -> AssetSpec:
```

Comment:
Should we do something [similar to PowerBI](https://github.com/dagster-io/dagster/blob/1.8.12/python_modules/libraries/dagster-powerbi/dagster_powerbi/translator.py#L126-L136) instead? 

I wonder if a function with the signature `def get_asset_spec(self, data: Union[SigmaDataset, SigmaWorkbook]) -> AssetSpec` would be easier to customize for users. We could call `get_asset_key` in it instead.

Reply:
Good question - @schrockn suggested that we put the `asset_key` as a parameter so that it's more clear that there is a contract that the asset key in the spec emitted from `get_asset_spec` must match the asset key returned by `get_asset_key`.

The integration itself would call `get_asset_key` and pass the result into `get_asset_spec`, e.g.

```python
asset_specs = [get_asset_spec(get_asset_key(obj), obj) for obj in objs]
```

I don't have a strong preference either way

Reply:
Thanks for clarifying. 

I'd be in favor of keeping the pattern without the key as a parameter for now, to ensure consistency between integrations. 

If we think that passing the key as a parameter to `get_asset_spec` is better, we should do it for all other translators.

Reply:
Updated to not pass the key for now